packages/salesforcedx-vscode-core/src/commands/util/betaDeployRetrieve.ts

```diff
@@ -4,37 +4,23 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-
-import { ComponentSet, MetadataType } from '@salesforce/source-deploy-retrieve';
+import { getRelativeProjectPath } from '@salesforce/salesforcedx-utils-vscode/out/src';
 import { MetadataComponent } from '@salesforce/source-deploy-retrieve';
-import * as vscode from 'vscode';
-import { sfdxCoreSettings } from '../../settings';
-
-export function useBetaDeployRetrieve(
-  uris: vscode.Uri[],
-  supportedTypes?: MetadataType[]
-): boolean {
-  const betaSettingOn = sfdxCoreSettings.getBetaDeployRetrieve();
-  if (!betaSettingOn) {
-    return false;
-  }
+import { SfdxPackageDirectories } from '../../sfdxProject';
 
-  const ws = new ComponentSet();
-  const permittedTypeNames = new Set();
-  supportedTypes?.forEach(type => permittedTypeNames.add(type.name));
+export async function formatException(e: Error): Promise<Error> {
+  const formattedException = new Error('Unknown Exception');
+  formattedException.name = e.name;
 
-  for (const { fsPath } of uris) {
-    const componentsForPath = ws.resolveSourceComponents(fsPath);
-    if (supportedTypes && componentsForPath) {
-      for (const component of componentsForPath) {
-        if (!permittedTypeNames.has(component.type.name)) {
-          return false;
-        }
-      }
-    }
+  if (e.name === 'TypeInferenceError') {
+    const projectPath = getRelativeProjectPath(
+      e.message.slice(0, e.message.lastIndexOf(':')),
+      await SfdxPackageDirectories.getPackageDirectoryPaths()
+    );
+    formattedException.message = `${projectPath}: Could not infer metadata type`;
   }
 
-  return true;
+  return formattedException;
```

Comment:
Convert to relative project path if it's a TypeInferenceError. Otherwise only throw the exception name with "Unknown Exception" as message.